raiden_contracts/data/source/services/MonitoringService.sol

```diff
@@ -177,8 +177,13 @@ contract MonitoringService is Utils {
     )
         public
     {
-        // Call updateReward first to avoid reentrancy problems when the
-        // token_network_address belongs to a malicious contract.
+        // Here we're trying to do bookkeeping first, but updateReward() first calls
+        // token_network_address.  So reentrancy is possible.
+        // In that case, the outer frame fails and reverts the state
+        // because token_network_address is not registered in the token_network_registry.
+        //
+        // Maybe it's simpler and safer to take the token address as an argument instead,
+        // and ask the TokenNetworkRegistry for the token_network_address.
```

Comment:
I don't see why it would be easier, but if we rely on registering networks for security, we definitely should check the registration.

Reply:
If we deploy this contract with care, we can make sure `token_network_registry.token_to_token_networks(token_network_token)`, causes no reentrancies.

In the current code, when we call token_network_address, we tolerate the possibility of reentrancy.